experiments/AMIP/user_io/ci_plots.jl

```diff
@@ -0,0 +1,203 @@
+                LABEL_LENGTH = 40
+                path = convert(Vector{Float64}, path)
+                normalized_path = lpad(path, LABEL_LENGTH + 1, " ")[(end - LABEL_LENGTH):end]
+
+                CairoMakie.Label(fig[0, col], path)
+            end
+        end
+        return fig
+    end
+
+    # Standardizes grid layout
+    gridlayout() =
+        map(1:MAX_PLOTS_PER_PAGE) do i
+            row = mod(div(i - 1, MAX_NUM_COLS), MAX_NUM_ROWS) + 1
+            col = mod(i - 1, MAX_NUM_COLS) + 1
+            return fig[row, col] = CairoMakie.GridLayout()
+        end
+
+    fig = makefig()
+    grid = gridlayout()
+    page = 1
+    grid_pos = 1
+
+    for var in vars
+        if grid_pos > MAX_PLOTS_PER_PAGE
+            fig = makefig()
+            grid = gridlayout()
+            grid_pos = 1
+        end
+
+        plot_fn(grid[grid_pos], var)
+        grid_pos += 1
+
+        # Flush current page
+        if grid_pos > min(MAX_PLOTS_PER_PAGE, vars_left_to_plot)
+            file_path = joinpath(save_path, "$(output_name)_$page.pdf")
+            CairoMakie.resize_to_layout!(fig)
+            CairoMakie.save(file_path, fig)
+            push!(summary_files, file_path)
+            vars_left_to_plot -= MAX_PLOTS_PER_PAGE
+            page += 1
+        end
+    end
+
+    # Save plots
+    output_file = joinpath(save_path, "$(output_name).pdf")
+
+    pdfunite() do unite
+        run(Cmd([unite, summary_files..., output_file]))
+    end
+
+    # Cleanup
+    Filesystem.rm.(summary_files, force = true)
+    return output_file
+end
+
+function map_comparison(func, simdirs, args)
+    return vcat([[func(simdir, arg) for simdir in simdirs] for arg in args]...)
+end
+
+"""
+    make_plots(
+        ::Union{Val{:general_ci_plots}},
+        output_paths::Vector{<:AbstractString},
+        plot_path::AbstractString;
+        reduction::String = "average",
+    )
+Create plots for the general CI diagnostics. The plots are saved to `plot_path`.
+This is the default plotting function for the CI diagnostics and it can be extended
+to include additional diagnostics.
+The `reduction` keyword argument should be consistent with the reduction used to save the diagnostics.
+"""
+function make_plots(
+    ::Union{Val{:general_ci_plots}},
+    output_paths::Vector{<:AbstractString},
+    plot_path::AbstractString;
+    reduction::String = "average",
+)
+    simdirs = SimDir.(output_paths)
+
+    # Default output diagnostics
+    short_names_3D = ["mse", "lr", "ediff"]
+    short_names_2D = ["ts"]
+
+    available_periods = ClimaAnalysis.available_periods(simdirs[1]; short_name = short_names_3D[1], reduction)
+    period = ""
+    if "10d" in available_periods
+        period = "10d"
+    elseif "1d" in available_periods
+        period = "1d"
+    elseif "12h" in available_periods
+        period = "12h"
+    end
+
+    available_periods = ClimaAnalysis.available_periods(simdirs[1]; short_name = short_names_2D[1], reduction)
```

Comment:
It seems like this line is duplicated from line 177

Reply:
this one is for [2d](url) but it was in the wrong place anyway - my bad! ;)